src/plugins/discover/public/application/main/components/layout/selected_vs_available_callout.tsx

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiCallOut } from '@elastic/eui';
+import type { DatatableColumn } from '@kbn/expressions-plugin/common';
+
+interface SelectedVSAvailableCallout {
+  isPlainRecord: boolean;
+  selectedColumns: string[];
+  textBasedQueryColumns?: DatatableColumn[];
+}
+
+export const SelectedVSAvailableCallout = ({
+  isPlainRecord,
+  textBasedQueryColumns,
+  selectedColumns,
+}: SelectedVSAvailableCallout) => {
+  return (
+    <>
+      {isPlainRecord &&
+        textBasedQueryColumns &&
+        selectedColumns.length > 0 &&
+        selectedColumns.length < textBasedQueryColumns.length && (
+          <EuiCallOut
+            color="primary"
+            data-test-subj="dscSelectedColumnsCallout"
+            iconType="iInCircle"
+            title={i18n.translate('discover.textBasedMode.selectedColumnsCallout', {
+              defaultMessage:
+                'Displaying first {selectedColumnsNumber} fields out of {textBasedQueryColumnsNumber}. You can add more fields to this table using Available fields of the Field list.',
```

Comment:
I think this copy could still use a bit of refinement. Maybe someone from @elastic/ux-writers could give us a hand?

Reply:
Asked @amyjtechwriter in slack! I will update when I have the feedback

Reply:
Synced with Amy offline. She suggested this

<img width="1954" alt="image" src="https://github.com/elastic/kibana/assets/17003240/e93e0686-efa5-4921-a1fe-a64486f46a12">

done https://github.com/elastic/kibana/pull/165557/commits/26677afe251357d0edbdfdf6f7e52960a9652d05